File: Dart/container-with-most-water.dart
// function to get most water inside ther container
int maxArea(List<int> height) {
  // 2 pointers
  int left = 0;
  int right = height.length - 1;
  int maxWater = 0;

  while (left < right) {
    int width = right - left;
    int h = height[left] < height[right] ? height[left] : height[right];
    int area = width * h;
    if (area > maxWater) maxWater = area;

    if (height[left] < height[right]) {
      left++;
    } else {
      right--;
    }
  }

  return maxWater;
}

void main() {
  print(maxArea([1, 8, 6, 2, 5, 4, 8, 3, 7])); // Output: 49
  print(maxArea([1, 1])); // Output: 1
}

File: Javascript/container-with-most-water.js
// funtion to get the most water inside the container
var maxArea = function (height) {
    // Two pointers
    let left = 0;
    let right = height.length - 1;
    let maxWater = 0;
    while (left < right) {
        // Width of the container
        const width = right - left;
        // Height is the smaller of the two lines
        const h = Math.min(height[left], height[right]);
        // Area calculation
        const area = width * h;
        maxWater = Math.max(maxWater, area);

        // Move the pointer pointing to the shorter line
        if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }
    return maxWater;
};

// Example
console.log(maxArea([1, 8, 6, 2, 5, 4, 8, 3, 7])); // Output: 49
console.log(maxArea([1, 1]));             // Output: 1